### ISSUE TITLE
Scrub scanning phase bandwidth calculation should take end of scan phase into account

### ISSUE THREAD

**Issue by GregorKopka** [openzfs/zfs#13090]
### System information

Type | Version/Name
 --- | ---
Distribution Name	| Funtoo
Distribution Version	| rolling
Kernel Version	| 4.19.194_p3-debian-sources-lts
Architecture	| x64 AMD Epyc
OpenZFS Version	| 2.0.6-r0

### Describe the problem you're observing
Running a scrub the bandwidth for the scan phase goes down over the whole time the scrub is running - even after the scan phase has already completed (only the 'issue'-Phase is still running).

This makes this figure somewhat pointless, IMHO.
Possibly the scan phase could track the time of it being actually active and calculate the bandwidth figure based on that?

### Describe how to reproduce the problem
`zpool scrub pool ; watch zpool status`

### Include any warning/errors/backtraces from the system logs
none

**Comment by GregorKopka** [comment]
Bad bot.

**Comment by behlendorf** [comment]
There was a recent change, #14410, made to the master branch which should help with this.  Any feedback would be welcome. This is something I'd like to backport for the next point release.

**Comment by GregorKopka** [comment]
Not really... as can be seen in the 'After'-Figure of the pull request, where the scan phase is still reported as 0B/s:
```
  scan: resilver in progress since Fri Jan 20 15:09:03 2023
        33.1T scanned at 0B/s, 29.0T issued at 2.26G/s, 33.1T total
        37.8G resilvered, 87.57% done, 00:31:01 to go
 ```

**Comment by behlendorf** [comment]
I see what you're saying.  It's reporting a current "scanning" rather than A "scanned" rate which is misleading.

**Comment by amotin** [comment]
I was going to look on this too, but haven't so far.  In my understanding ZFS does not account separately time spent on scanning and issuing.  And since those two may switch multiple times in process, resulted rates are not very meaningful.  It probably needs some time accounting changes.

**Comment by amotin** [comment]
This should improve the reporting: https://github.com/openzfs/zfs/pull/15007.  It won't report speed for completed scan phase and same time report more reasonable issue speeds.  I was trying to account separately the scan and issue times as I planned, and mostly implemented it, but I found that it does not help much with the scrub time forecasting, so I de decided it does not worth it.
```
  scan: resilver in progress since Thu Jun 22 14:03:47 2023
        625G / 625G scanned, 1.54G / 41.6G issued at 112M/s
        1.52G resilvered, 3.69% done, 00:06:05 to go
```

**Comment by GregorKopka** [comment]
When dropping the scan speed on completion, why not also drop the x / x display for it?

**Comment by amotin** [comment]
User-space does not really know that scan is complete.  Speeds are hidden when there is no activity in this pass, and it is kernel that opens a new pass when scan is complete, effectively hiding the scan speed.
